### PR TITLE
CodeQL docs: tidy up 'docs-preparation' branch following pre-migration work

### PR DESCRIPTION
**Pull request by jf205**
Fixes https://github.com/github/semmle-docs/issues/1.

### Summary

This PR makes a few updates to the `docs-preparation` branch following the first round of content changes. Hopefully after this we will be able to merge `docs-preparation` into `master` and make some further changes that require more substantial content changes (starting with https://github.com/github/product-documentation/issues/2159).

### Changes

There are quite a few changes in this PR, so I'll illustrate with a few examples.

Change | Example before | Example after
---|---|---
'CodeQL for X' map topics manually updated to include a TOC with article names and introductions. 'Other resources' renamed to 'Further reading' | [CodeQL for Java](https://help.semmle.com/QL/learn-ql/java/ql-for-java.html) | [CodeQL for Java](http://docteam.internal.semmle.com/james/docs-preparation-branch/learn-ql/java/ql-for-java.html)
Consistent intros for the 'CodeQL for X' map topics | [CodeQL for JavaScript](https://help.semmle.com/QL/learn-ql/javascript/ql-for-javascript.html) | [CodeQL for JavaScript](http://docteam.internal.semmle.com/james/docs-preparation-branch/learn-ql/javascript/ql-for-javascript.html)
'QL tutorials' map topic manually updated to include a TOC with article names and introductions. | [QL detective tutorials](https://help.semmle.com/QL/learn-ql/beginner/ql-tutorials.html) | [QL tutorials](http://docteam.internal.semmle.com/james/docs-preparation-branch/learn-ql/beginner/ql-tutorials.html)
'CodeQL queries' map topics manually updated to include a TOC with article names and introductions. | [Writing CodeQL queries](https://help.semmle.com/QL/learn-ql/writing-queries/writing-queries.html) | [CodeQL queries](http://docteam.internal.semmle.com/james/docs-preparation-branch/learn-ql/writing-queries/writing-queries.html)
'What next?' (or similar) -> 'Further reading' | [What next?](https://help.semmle.com/QL/learn-ql/cpp/function-classes.html#what-next) | [Further reading](http://docteam.internal.semmle.com/james/docs-preparation-branch/learn-ql/cpp/function-classes.html#further-reading)
Remove link to a section that no longer exists. | [CodeQL training, More resources](https://help.semmle.com/QL/learn-ql/ql-training.html#more-resources) | [CodeQL training, Further reading](http://docteam.internal.semmle.com/james/docs-preparation-branch/learn-ql/ql-training.html#further-reading)
'Learning CodeQL' landing page simplified so it's more similar to the GitHub landing page. 'Learning CodeQL' won't be migrated, so I didn't spend too long on this. | [Learning CodeQL](https://help.semmle.com/QL/learn-ql/index.html) | [Learning CodeQL](http://docteam.internal.semmle.com/james/docs-preparation-branch/learn-ql/index.html)

### Notes

- I've updated the titles of the 'Further reading' sections, but not made any content changes. This is something that would be good to discuss later.
- I've not made nay changes to the content in the 'CodeQL training' section. We are waiting for some input from product here.
- [CodeQL for COBOL](http://docteam.internal.semmle.com/james/docs-preparation-branch/learn-ql/cobol/ql-for-cobol.html) still exists on this branch. It will disappear when we merge into master.
- [CodeQL for Go](http://docteam.internal.semmle.com/james/docs-preparation-branch/learn-ql/go/ql-for-go.html) is still just a single link to the LGTM help. Max has created a topic about the CodeQL library for Go, but I don't think we haven't worked out a way of publishing it yet--@shati-patel, what's the plan for that topic?

### CODE REVIEW

File: docs/language/learn-ql/python/ql-for-python.rst [hunk 1]
```diff
@@ -14,3 +13,23 @@ Experiment and learn how to write effective and efficient queries for CodeQL dat
    control-flow
    taint-tracking
 
+-  `Basic Python query <https://lgtm.com/help/lgtm/console/ql-python-basic-example>`__ : Learn to write and run a simple CodeQL query using LGTM.
+
+-  :doc:`CodeQL library for Python <introduce-libraries-python>`: When you need to analyze a Python program, you can make use of the large collection of classes in the CodeQL library for Python.
+
+-  :doc:`Functions in Python <functions>`: You can use syntactic classes from the standard CodeQL library to find Python functions and identify calls to them.
+
+-  :doc:`Expressions and statements in Python <statements-expressions>`: You can use syntactic classes from the CodeQL library to explore how Python expressions and statements are used in a code base.
```
Comment (hubwriter) [inline docs/language/learn-ql/python/ql-for-python.rst]:
```suggestion
-  :doc:`Expressions and statements in Python <statements-expressions>`: You can use syntactic classes from the CodeQL library to explore how Python expressions and statements are used in a codebase.
```

**Comment by hubwriter** [comment]
👍 💛

**Comment by shati-patel** [comment]
> [CodeQL for Go](http://docteam.internal.semmle.com/james/docs-preparation-branch/learn-ql/go/ql-for-go.html) is still just a single link to the LGTM help. Max has created a topic about the CodeQL library for Go, but I don't think we have worked out a way of publishing it yet--@shati-patel, what's the plan for that topic?

Good point. In https://github.com/github/codeql-go/pull/25 Max mentioned hooking up the Go docs to the Sphinx build, but it's probably not worth making changes to the Sphinx build any more. 
@max-schaefer, would you be okay with us moving the Go library overview into this repo alongside the other docs? Since we're tidying things up for when we move the Semmle docs over to GitHub, it would be helpful to have everything in one place (even if that means separating the Go docs from the Go code...)

**Comment by jf205** [comment]
Thanks for the review @hubwriter!

**Comment by jf205** [comment]
> Hopefully after this we will be able to merge `docs-preparation` into `master`

I forgot that there is at least one other issue that we need to fix before we merge `docs-preparation` into `master`: https://github.com/github/semmle-docs/issues/21.

**Comment by max-schaefer** [comment]
> @max-schaefer, would you be okay with us moving the Go library overview into this repo alongside the other docs?

I suppose that's fine. It will be a little harder to keep in sync, but we'll manage.

**Pull request merged**